return {
    OnBoatProductPurchase = function() --[[LuaEncode: `options.FunctionsReturnRaw` false; can't serialize functions]] return end,
    Spawn = function() --[[LuaEncode: `options.FunctionsReturnRaw` false; can't serialize functions]] return end,
    library = {
        ["Holiday Themed - Speedboat"] = {
            Dedscriptipon = "Sail around at unpresedented speeds! HOLIDAY XMAS!",
            BackwardsEfficiency = 1,
            Price = -1,
            Bobbing = 2,
            BobbingSpeed = 1,
            MaxSpeed = 150,
            StopEfficiency = 1.5,
            Level = 1,
            TurningSpeed = 0.54,
            Accel = 0.3
        },
        ["Holiday Steampunk Ship"] = {
            ExpirationDate = DateTime.fromUnixTimestamp(1735398000),
            BackwardsEfficiency = 1.1,
            Price = -1,
            Bobbing = 2.5,
            StopEfficiency = 2.2,
            BobbingSpeed = 3,
            TurningSpeed = 0.33,
            MaxSpeed = 100,
            ProductId = 2677594586,
            Dedscriptipon = "Holiday Steampunk Ship",
            Level = 1,
            Icon = "rbxassetid://90047369440873",
            Accel = 0.25
        },
        ["Snowmobile Jetski"] = {
            Dedscriptipon = "Jetski frozen christmas version!",
            BackwardsEfficiency = 1,
            Price = -1,
            Bobbing = 2,
            BobbingSpeed = 0.4,
            MaxSpeed = 200,
            StopEfficiency = 3.5,
            TurningSpeed = 0.7,
            Level = 60,
            Accel = 0.3,
            ObtainText = "Obtain on Christmas Bundle"
        },
        ["Archaeological Boat"] = {
            Dedscriptipon = "Boat used by archaeologists from ancient island.",
            BackwardsEfficiency = 1,
            Price = 20000,
            Bobbing = 1,
            BobbingSpeed = 3,
            MaxSpeed = 80,
            StopEfficiency = 2,
            Level = 30,
            TurningSpeed = 0.3,
            Accel = 0.2
        },
        Jetski = {
            Dedscriptipon = "Ski around with a friend at insane speeds!",
            BackwardsEfficiency = 1,
            Price = 50000,
            Bobbing = 2,
            BobbingSpeed = 0.4,
            MaxSpeed = 180,
            StopEfficiency = 3.5,
            Level = 60,
            TurningSpeed = 0.7,
            Accel = 0.3
        },
        ["Coral Cruiser Boat"] = {
            Dedscriptipon = "Fischer Special Special!",
            BackwardsEfficiency = 1,
            Price = -1,
            Bobbing = 2.5,
            BobbingSpeed = 1,
            StopEfficiency = 1.3,
            MaxSpeed = 85,
            TurningSpeed = 0.5,
            Accel = 0.25,
            Level = 0,
            Icon = "rbxassetid://71002238163754",
            ObtainText = "Obtain on Starter Pack"
        },
        Rowboat = {
            Dedscriptipon = "Sail the seas with your trusty paddles!",
            BackwardsEfficiency = 1,
            Price = 700,
            Bobbing = 3,
            BobbingSpeed = 1,
            MaxSpeed = 25,
            StopEfficiency = 2,
            Level = 0,
            TurningSpeed = 0.4,
            Accel = 0.15
        },
        ["The Pearl"] = {
            Dedscriptipon = "No one knows what sails this ship...",
            BackwardsEfficiency = 1,
            Price = 50000,
            Bobbing = 6,
            BobbingSpeed = 3,
            MaxSpeed = 95,
            StopEfficiency = 3,
            Level = 90,
            TurningSpeed = 0.3,
            Accel = 0.2
        },
        ["Flying Dutchman"] = {
            Dedscriptipon = "Only available during FischFright-2024",
            BackwardsEfficiency = 1,
            Price = -1,
            Bobbing = 6,
            BobbingSpeed = 3,
            MaxSpeed = 80,
            StopEfficiency = 2,
            Level = 0,
            TurningSpeed = 0.3,
            Accel = 0.2
        },
        ["Reindeer Cruiser"] = {
            Dedscriptipon = "A ducky boat with a pretty convincing reindeer disguise. [Only obtainable during Fischmas 2024]",
            BackwardsEfficiency = 1,
            Price = -1,
            Bobbing = 4,
            BobbingSpeed = 0.5,
            MaxSpeed = 55,
            StopEfficiency = 2,
            Level = 1,
            TurningSpeed = 0.5,
            Accel = 0.25
        },
        Pontoon = {
            Dedscriptipon = "Great for parties with some buddies!",
            BackwardsEfficiency = 1,
            Price = 20000,
            Bobbing = 1,
            BobbingSpeed = 1,
            MaxSpeed = 40,
            StopEfficiency = 2,
            Level = 30,
            TurningSpeed = 0.4,
            Accel = 0.2
        },
        Sleigh = {
            Dedscriptipon = "Deliver presents at ridiculous speeds! [Only obtainable during Fischmas 2024]",
            BackwardsEfficiency = 0.85,
            Price = -1,
            Bobbing = 2,
            BobbingSpeed = 1,
            MaxSpeed = 90,
            StopEfficiency = 1.5,
            Level = 1,
            TurningSpeed = 0.6,
            Accel = 0.4
        },
        ["Quack Cruiser"] = {
            Dedscriptipon = "This duck sailed up to the lemonade stand.",
            BackwardsEfficiency = 1,
            Price = 5000,
            Bobbing = 4,
            BobbingSpeed = 0.5,
            MaxSpeed = 45,
            StopEfficiency = 2,
            Level = 35,
            TurningSpeed = 0.5,
            Accel = 0.2
        },
        Speedboat = {
            Dedscriptipon = "Sail around at remarkable speeds!",
            BackwardsEfficiency = 1,
            Price = 15000,
            Bobbing = 2.5,
            BobbingSpeed = 1,
            MaxSpeed = 85,
            StopEfficiency = 1.3,
            Level = 20,
            TurningSpeed = 0.5,
            Accel = 0.25
        },
        Titan = {
            Dedscriptipon = "Huh? Who would want this?",
            BackwardsEfficiency = 1,
            Price = 5000,
            Bobbing = 3,
            BobbingSpeed = 1,
            MaxSpeed = 45,
            StopEfficiency = 1.5,
            Level = 40,
            TurningSpeed = 0.2,
            Accel = 0.2
        },
        ["Fishing Boat"] = {
            Dedscriptipon = "A trusty boat to reel in even the biggest fish!",
            BackwardsEfficiency = 1,
            Price = 20000,
            Bobbing = 2,
            BobbingSpeed = 1.5,
            MaxSpeed = 75,
            StopEfficiency = 2,
            Level = 25,
            TurningSpeed = 0.5,
            Accel = 0.3
        },
        ["Santa\xE2\x80\x99s Jetski"] = {
            ExpirationDate = DateTime.fromUnixTimestamp(1735398000),
            BackwardsEfficiency = 1.1,
            Price = -1,
            Bobbing = 2,
            TurningSpeed = 0.8,
            StopEfficiency = 3.7,
            BobbingSpeed = 0.4,
            Dedscriptipon = "Santa\xE2\x80\x99s Jetski christmas version!",
            MaxSpeed = 220,
            ProductId = 2677594486,
            LimitedAmount = 20000,
            Level = 1,
            Icon = "rbxassetid://140332102870016",
            Accel = 0.4
        },
        ["Pirate Ship"] = {
            Dedscriptipon = "Argggg Matey!!",
            BackwardsEfficiency = 1,
            Price = 20000,
            Bobbing = 6,
            BobbingSpeed = 3,
            MaxSpeed = 80,
            StopEfficiency = 2,
            Level = 30,
            TurningSpeed = 0.3,
            Accel = 0.2
        },
        ["Luxury Speedboat"] = {
            Dedscriptipon = "Sail around at unpresedented speeds!",
            BackwardsEfficiency = 1,
            Price = 20000,
            Bobbing = 2,
            BobbingSpeed = 1,
            MaxSpeed = 150,
            StopEfficiency = 1.5,
            Level = 50,
            TurningSpeed = 0.54,
            Accel = 0.3
        },
        Surfboard = {
            Dedscriptipon = "Surf the waves, catch the sharks!",
            BackwardsEfficiency = 1,
            Price = 1500,
            Bobbing = 3,
            BobbingSpeed = 0.5,
            MaxSpeed = 30,
            StopEfficiency = 2,
            Level = 7,
            TurningSpeed = 0.5,
            Accel = 0.2
        },
        ["Fischmas Speedboat"] = {
            ExpirationDate = DateTime.fromUnixTimestamp(1735398000),
            BackwardsEfficiency = 1.1,
            Price = -1,
            Bobbing = 2,
            StopEfficiency = 1.8,
            BobbingSpeed = 1,
            TurningSpeed = 0.6,
            MaxSpeed = 180,
            ProductId = 2677594734,
            Dedscriptipon = "The Fischmas Speedboat!",
            Level = 1,
            Icon = "rbxassetid://130906482985374",
            Accel = 0.35
        },
        Hovercraft = {
            Dedscriptipon = "Hover around the seas! [Submission; @DDkreep, @LiamGame09, @kylecat11]",
            BackwardsEfficiency = 1,
            Price = 7000,
            Bobbing = 1,
            BobbingSpeed = 1,
            MaxSpeed = 55,
            StopEfficiency = 0.7,
            Level = 15,
            TurningSpeed = 0.4,
            Accel = 0.22
        },
        ["Bass Boat"] = {
            Dedscriptipon = "Great for fishing with pals!",
            BackwardsEfficiency = 1,
            Price = 6000,
            Bobbing = 2,
            BobbingSpeed = 1,
            MaxSpeed = 45,
            StopEfficiency = 0.7,
            Level = 10,
            TurningSpeed = 0.4,
            Accel = 0.22
        }
    },
    Give = function() --[[LuaEncode: `options.FunctionsReturnRaw` false; can't serialize functions]] return end
}